Add gRPC icon

0x6a616e opened the pull request:
![grpc](https://github.com/simple-icons/simple-icons/assets/48343109/2f4679e9-ffbd-4ad8-bab8-5a9ef935c912)

**Issue:** closes #6275

**Similarweb rank:** https://www.similarweb.com/website/grpc.io/

### Checklist

- [X] I updated the JSON data in `_data/simple-icons.json`
- [X] I optimized the icon with SVGO or SVGOMG
- [X] The SVG `viewbox` is `0 0 24 24`

### Description

For the SVG I used the logo they have in the GitHub repository for their web page, I also took the HEX color from said page, from the navigation bar. I wasn't completely sure about the brand guidelines, I followed the Trademark section in the page.

PeterShaggyNoble reviewed (changes requested):
Thanks for the contribution, @0x6a616e 👍🏻 I'm seeing some small differences throughout when comparing with source, with [this one](https://github.com/simple-icons/simple-icons/assets/15157491/2b6ceebd-a683-4e4f-8632-dc2a0d387255) from the top of the "c" being one of the more pronounced ones (yours in red). Part of this could be caused by the fact that your version is squashed vertically; the original scales down to `9.738` high, while yours is `9.731`.

Would you mind having another look to see if you could fix up differences?

0x6a616e commented:
![grpc](https://github.com/simple-icons/simple-icons/assets/48343109/e9d5725d-e442-486b-9436-05775b4d8604)

I modified the svg to fix the mentioned squashing, I played with it to try and reduce the differences when compared to the original but I'm not completely sure if I fixed all of them.

PeterShaggyNoble reviewed (changes requested):
That lines up perfectly now, @0x6a616e. However, there are a lot of excess points in the letters. Would you be able to try removing some of them, please, taking care not to introduce any differences with the source? Apologies, I should have caught that in my initial review.
